tests/e2e/specs/out-of-game/home.spec.js

```diff
@@ -29,6 +30,15 @@ function toggleInput(selector, checked = false) {
     .should(after);
 }
 
+function recoverOpponentsInNewGame(gameId) {
+  cy.recoverSessionOpponent(playerOne);
+  cy.subscribeOpponent(gameId);
+  cy.readyOpponent(gameId);
+  cy.recoverSessionOpponent(playerTwo);
+  cy.subscribeOpponent(gameId);
+  cy.readyOpponent(gameId);
+}
```

Comment:
This is a handy utility and it looks like it gets a fair bit of use here. It looks like it's only used immediately after creating a game, so it likely makes sense to combine this with the game creation itself. The only issue with doing that is how to get access to the `gameId` of the newly created game, since that's provided by the `createGameOpponent()`. Here we can use [Cypress aliases](https://docs.cypress.io/guides/core-concepts/variables-and-aliases) to capture the reference to the gameId.

```suggestion
function setupGameBetweenTwoUnseenPlayers(gameName) {
  cy.createGameOpponent(gameName).then(({ gameId }) => {
    cy.wrap(gameId).as(`${gameName}GameId`); // this creates an alias for the gameId
    cy.recoverSessionOpponent(playerOne);
    cy.subscribeOpponent(gameId);
    cy.readyOpponent(gameId);
    cy.recoverSessionOpponent(playerTwo);
    cy.subscribeOpponent(gameId);
    cy.readyOpponent(gameId);
    });
}
```

Then you can use it in tests like so:
```
setupGameBetweenTwoUnseenPlayers('conceded');
cy.concedeOpponent();
cy.get(`[data-cy-spectate-game=${this.concededGameId}]`).should('be.disabled');
```

One gotcha is that the reference to `this.concededGameId` only works when the test's callback uses the actual `function` keyword instead of an arrow function `() = >{}`. So we'd have to update the test like so:

```
it('Disables spectate button if on home view before game finishes', function () {
  // test code
});
```

Vs the current arrow function wouldn't allow the reference to `this.concededGameId`

```
it('Does not work without the function keyword', () => {
  // this won't work because arrow functions treat `this` differently
});
```

This might simplify the consumers of the function here since they all involve creating the game in addition to joining it as the other players